### ISSUE TITLE
Naga frontend supports gles

### ISSUE THREAD

**Issue by yorkie** [gfx-rs/wgpu#6335]
**Is your feature request related to a problem? Please describe.**
I'm looking for a library to modify the glsl(gles 300/310) source via `Module`, but the naga seems not supporting the ES profile.

**Describe the solution you'd like**
Support the ES profile at glsl frontend?

**Comment by cwfitzgerald** [comment]
I don't think we have any plans on doing this. The GLSL frontend is already barely maintained, so having it support both GL and Vulkan flavored GLSL just isn't feasible.
